### PR TITLE
Add new INTER system test

### PR DESCRIPTION
**Pull request by gemmaguest**
**Description of work.**
Add a new system test for standard reflectometry reduction on INTER

**Report to:** Max at ISIS

**To test:**
Code review and check that the test works

*There is no associated issue.*

*This does not require release notes* because it is just adding a system test

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Testing/SystemTests/tests/analysis/INTERReductionTest.py [hunk 1]
```diff
@@ -0,0 +1,349 @@
...
+        SaveNexus(InputWorkspace=INTERReductionTest.result_workspace,
+                  Filename=INTERReductionTest.reference_result_file)
+
+    @staticmethod
+    def regenerateReferenceFileFromDirectory(reference_file_directory):
+        SetupInstrument()
+        RegenerateReferenceFile(reference_file_directory, INTERReductionTest.reference_result_file)
+
+
+def SetupInstrument():
```
Comment (DanNixon) [inline Testing/SystemTests/tests/analysis/INTERReductionTest.py]: Use `camalCase` (more "Mantid-ish") or `snake_case` (more "Python-ish") for function names.

File: Testing/SystemTests/tests/analysis/INTERReductionTest.py [hunk 2]
```diff
@@ -0,0 +1,349 @@
...
+            SecondTransmissionRun=run2,
+            OutputWorkspace=name,
+            StartOverlap=10,
+            EndOverlap=12)
+        # Delete debug workspaces
+        DeleteWorkspace('TRANS_LAM_'+run1)
+        DeleteWorkspace('TRANS_LAM_'+run2)
+
+
+def EventRef(run_number,angle,start=0,stop=0,DB='TRANS'):
```
Comment (DanNixon) [inline Testing/SystemTests/tests/analysis/INTERReductionTest.py]: Space after `,` for parameters.

File: Testing/SystemTests/tests/analysis/INTERReductionTest.py [hunk 3]
```diff
@@ -0,0 +1,349 @@
...
+    filter_ws_name=run_name+'_filter'
+    FilterByTime(InputWorkspace=run_name, OutputWorkspace=filter_ws_name, StartTime=start, StopTime=stoptime)
+    # Calculate the fraction of proton charge in this slice
+    filter_workspace=mtd[filter_ws_name]
+    slice_proton_charge = filter_workspace.getRun().getLogData('gd_prtn_chrg').value
+    total_proton_charge = run_workspace.getRun().getLogData('gd_prtn_chrg').value
+    fraction = slice_proton_charge/total_proton_charge
+    duration = filter_workspace.getRun().getLogData('duration').value
+    print('Fraction:',fraction)
+    print('Slice:',slice)
```
Comment (DanNixon) [inline Testing/SystemTests/tests/analysis/INTERReductionTest.py]: Is `slice` a variable in this context?

**Review by DanNixon — CHANGES_REQUESTED** [review]
Test looks good. Ideally the functions could do with a brief high level description of what specifically they are doing (to avoid having to read the entire function to determine this).

The other comments are just style related.

File: Testing/SystemTests/tests/analysis/INTERReductionTest.py (hunk 3 above)
Comment (gemmaguest) [reply Testing/SystemTests/tests/analysis/INTERReductionTest.py]: Sorry, that got left in after it was renamed to `slice_proton_charge`. Thanks for spotting it!

**Review by DanNixon — APPROVED** [review]
Looks good to me now.

**Comment by martyngigg** [comment]
It was requested that this be reported to Max at ISIS when merged.

**Pull request merged**